poller/poller.go

```diff
@@ -0,0 +1,123 @@
+package poller
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"time"
+
+	"github.com/jackc/pgx/v4"
+	"github.com/likecoin/likechain/app"
+	"github.com/likecoin/likecoin-chain-tx-indexer/db"
+	"github.com/likecoin/likecoin-chain-tx-indexer/logger"
+	"github.com/tendermint/go-amino"
+	cmn "github.com/tendermint/tendermint/libs/common"
+	coreTypes "github.com/tendermint/tendermint/rpc/core/types"
+)
+
+func getResponse(client *http.Client, url string) ([]byte, error) {
+	resp, err := client.Get(url)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+	body, err := ioutil.ReadAll(resp.Body)
+	if err != nil {
+		return nil, err
+	}
+	return body, nil
+}
+
+type CosmosCallContext struct {
+	Codec       *amino.Codec
+	Client      *http.Client
+	LcdEndpoint string
+}
+
+func getBlock(ctx *CosmosCallContext, height int64) (*coreTypes.ResultBlock, error) {
+	heightStr := "latest"
+	if height > 0 {
+		heightStr = fmt.Sprintf("%d", height)
+	}
+	url := fmt.Sprintf("%s/blocks/%s", ctx.LcdEndpoint, heightStr)
+	body, err := getResponse(ctx.Client, url)
+	if err != nil {
+		return nil, err
+	}
+	resultBlock := coreTypes.ResultBlock{}
+	err = ctx.Codec.UnmarshalJSON(body, &resultBlock)
+	if err != nil {
+		return nil, err
+	}
+	return &resultBlock, nil
+}
+
+type TxResult struct {
+	Height int64 `json:"height"`
+	Logs   []struct {
+		Events []struct {
+			Type       string `json:"type"`
+			Attributes []struct {
+				Key   string `json:"key"`
+				Value string `json:"value"`
+			} `json:"attributes"`
+		} `json:"events"`
+	} `json:"logs"`
+}
+
+func Run(conn *pgx.Conn, lcdEndpoint string) {
+	if lcdEndpoint[len(lcdEndpoint)-1] == '/' {
+		lcdEndpoint = lcdEndpoint[:len(lcdEndpoint)-1]
+	}
+	ctx := CosmosCallContext{
+		Codec:       app.MakeCodec(),
+		Client:      &http.Client{},
+		LcdEndpoint: lcdEndpoint,
+	}
+
+	batchSize := 1000
+	batch := db.NewBatch(conn, batchSize)
+	dbHeight, err := db.GetLatestHeight(conn)
+	if err != nil {
+		// TODO: retry
+		logger.L.Panicw("Cannot get height from database", "error", err)
```

Comment:
prefer exit with error code instead of just log?

Reply:
`panic` in Go is similar to `throw` in other languages, which will crash the program if not caught (`recover` in Go).